core/src/avm2/globals/flash/display/displayobject.rs

```diff
@@ -682,6 +683,83 @@ pub fn set_blend_mode<'gc>(
     Ok(Value::Undefined)
 }
 
+fn scroll_rect<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(dobj) = this.and_then(|this| this.as_display_object()) {
+        if let Some(scroll_rect) = dobj.scroll_rect() {
+            return Ok(activation
+                .avm2()
+                .classes()
+                .rectangle
+                .construct(
+                    activation,
+                    &[
+                        scroll_rect.x_min.to_pixels().into(),
+                        scroll_rect.y_min.to_pixels().into(),
+                        (scroll_rect.x_max - scroll_rect.x_min).to_pixels().into(),
+                        (scroll_rect.y_max - scroll_rect.y_min).to_pixels().into(),
+                    ],
+                )?
+                .into());
+        } else {
+            return Ok(Value::Null);
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+fn set_scroll_rect<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(dobj) = this.and_then(|this| this.as_display_object()) {
+        let rect = args[0].as_object().unwrap();
```

Comment:
Not super important, but this accepts a normal `Object` with the given properties when it should only accept a `Rectangle`. This will probably get taken care of when we support specific argument types on our native functions.

Reply:
We can get this for free if we define `DisplayObject` in ActionScript, and make this a native method.